Consistently show bot icon after bot name #26831

anandshende opened the pull request:
**Description**:
Add bot icon in Compose Message after bot name.
Adding an **is_bot** property in **input_pill's** `create` method.

**Fixes**:
Showing bot icon after bot name in compose message - [Link](https://github.com/zulip/zulip/issues/26831)

**Screenshots**:

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html

**Screenshots and screen captures:**

![Compose Message{caption=Compose Message}](https://github.com/zulip/zulip/assets/24247490/8e22eaea-c7c2-4a1a-8b96-a3c3fe3be79c)
![Chat List{caption=Chat List}](https://github.com/zulip/zulip/assets/24247490/23333ec1-2ce0-4c14-a1b8-09d5ba6fc51f)
![Message List{caption=Message List}](https://github.com/zulip/zulip/assets/24247490/88e3f1bb-6fdf-4ce3-b50c-9fcfc0e85cef)

<details>
<summary>Self-review checklist</summary>


- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

anandshende commented:
“Ready for review!”

alya commented:
Thanks! The icon placement doesn't look right to me in the compose pill. It seems a bit too big, too low, and too close to the name.

alya commented:
Once fixed, please include a light theme screenshot as well.

anandshende commented:
![image](https://github.com/zulip/zulip/assets/24247490/8653e130-e4e9-48ec-9a34-4b25efdb3d1d)
![image](https://github.com/zulip/zulip/assets/24247490/275f2d72-f242-4f53-8aad-30cf84c26474)

Hi @alya,
Updated the CSS properties and Light Theme screenshot as you suggested. Could you please review it again ?

alya commented:
That's looking better!

Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

alya commented:
Also, please add a screenshot with a very long bot name that gets abbreviated with `...`. We still want to show the icon in that case.

anandshende commented:
Hi @alya,

As requested, adding the screenshot with ellipsis(`...`) in the bot name.
![image](https://github.com/zulip/zulip/assets/24247490/88c40b75-29c4-44a7-b0bc-61c8f9a1ebe8)
Also cleaned the commit history.

Please review once again.

alya commented:
Thanks! @karlstolley are you up for reviewing this one? I haven't tested.

karlstolley commented on web/styles/components.css:
```diff
@@ -96,7 +96,13 @@ a.no-underline:hover {
     color: var(--color-icon-bot);
     vertical-align: top;
-    padding: 0 2px;
+    padding: 0 0 0 5px;
```
This likely affects every instance of `.zulip-icon-bot` in the interface; have you grepped to make sure that it's not breaking other presentations, e.g., in message headers?

karlstolley commented on web/styles/components.css:
```diff
@@ -96,7 +96,13 @@ a.no-underline:hover {
+}
+
+.pill-label {
+    & i.zulip-icon.zulip-icon-bot {
```
If it's possible, simplify this nested selector to just `.zulip-icon-bot`

anandshende replied:
updated...

karlstolley commented on web/templates/input_pill.hbs:
```diff
@@ -7,6 +7,9 @@
         {{~#if has_status~}}
             {{~> status_emoji status_emoji_info~}}
         {{~/if~}}
+        {{#if is_bot}}
```
Use tildes here (`~`) to collapse the whitespace, as in the `has_status` if statement above, and then you shouldn't have to modify the `search.test.js` file as you've done here.

anandshende replied:
Updated...

anandshende commented:
@karlstolley  - I have updated the code based on your comments.
I have cleaned the git history.
All test cases are passing.

I have been able to find the bot icon in these 3 places only.
![image](https://github.com/zulip/zulip/assets/24247490/2f130d01-9fc9-4fc7-a16d-8f6c7a25ad49)
![image](https://github.com/zulip/zulip/assets/24247490/1acbe9be-7e2c-48f6-8bae-45c8ee0ece7d)
![image](https://github.com/zulip/zulip/assets/24247490/e71389c8-6b16-4544-b854-1bff6a7dae5b)

alya commented:
I believe @karlstolley was requesting side-by-side before/after screenshots. You can use a table to make them easy to compare.

anandshende commented:
| Element | After | Before |
| -------- | -------- | ------- |
| Compose Message | ![image](https://github.com/zulip/zulip/assets/24247490/bfc95ab9-2ef8-45c8-9bad-35b7cf01a8e8) | ![image](https://github.com/zulip/zulip/assets/24247490/9d6981b8-d3ea-459c-961c-666ff5b4121b)  |
| Message List |  ![image](https://github.com/zulip/zulip/assets/24247490/07ffb302-773c-4ada-943d-227fc1cfe419) | ![image](https://github.com/zulip/zulip/assets/24247490/c05bb869-a3f6-47ad-913c-28a1c17eb588) |
| Chat List | ![image](https://github.com/zulip/zulip/assets/24247490/95345439-2bd3-4dcc-8e9e-2d6c2e49f212) | ![image](https://github.com/zulip/zulip/assets/24247490/5e94f8fa-ab05-43f1-97ee-91de56a640fa) | 


@alya - Thanks for clarifying.
@karlstolley - Please approve.

karlstolley commented:
I'd like to see the screenshots at the top of the PR; you should be able to edit your original content. Also, the before and after screen shots should have EXACTLY the same content. Your before & after set is all scattershot, so it's impossible to see what ought to be compared.

anandshende commented:
Hi All, I will not be able to contribute further on this. Therefore please consider this PR abandoned.

timabbott commented:
Closing, as the issue was fixed in https://github.com/zulip/zulip/pull/31894.
